spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraStrategies.scala

```diff
@@ -121,7 +121,8 @@ private[cassandra] trait CassandraStrategies {
         val eqPredicates = eqColumns.flatMap(eqPredicatesByName)
         val optionalNonEqPredicate = for {
           c <- otherColumns.headOption.toSeq
-          p <- firstNonEmptySeq(rangePredicatesByName(c), inPredicatesByName(c).filter(_ => eqColumns.nonEmpty))
+          p <- firstNonEmptySeq(rangePredicatesByName(c), inPredicatesByName(c).filter(
+            _ => eqColumns.nonEmpty && c==clusteringColumns.last))
```

Comment:
Well - I'm not sure - I thought that you can use `in` on the clustering column if eq predicates are defined on the preceding clustering columns. I tried this in cqlsh and got the following result:
```
cqlsh:ks> select * from users2 where cid2 in (4, 5) allow filtering;

InvalidRequest: code=2200 [Invalid query] message="PRIMARY KEY column "cid2" cannot be restricted (preceding column "cid1" is either not restricted or by a non-EQ relation)"
```